### PR TITLE
Update SDS integration tests

### PR DESCRIPTION
**Pull request by lei-tang**
- Use ServiceEntry in the Vault integration test.
- Remove redundant mesh policy and destination rule for mTLS, since global.mtls.enabled=true.

### CODE REVIEW

**Comment by lei-tang** [comment]
https://github.com/istio/istio/issues/13439

File: tests/integration/security/sds_citadel_flow/sds_citadel_flow_test.go [hunk 1]
```diff
@@ -85,16 +83,8 @@ func TestSdsCitadelCaFlow(t *testing.T) {
...
-				map[string]string{
-					"Namespace": ns.Name(),
-				})
-			g.ApplyConfigOrFail(t, systemNS, deployment)
-			defer g.DeleteConfigOrFail(t, systemNS, deployment)
-
-			// Sleep 3 seconds for the policy to take effect.
-			time.Sleep(3 * time.Second)
+			// Sleep 10 seconds for the policy to take effect.
+			time.Sleep(10 * time.Second)
```
Comment (incfly) [inline tests/integration/security/sds_citadel_flow/sds_citadel_flow_test.go]: any reason to increase the waiting time from 3 to 10 seconds?

File: tests/integration/security/sds_vault_flow/sds_vault_flow_test.go [hunk 2]
```diff
@@ -85,16 +85,17 @@ func TestSdsVaultCaFlow(t *testing.T) {
 				},
 			}
 
-			// Apply the policy to the system namespace.
-			deployment := tmpl.EvaluateOrFail(t, file.AsStringOrFail(t, "testdata/global-mtls.yaml"),
+			// Apply the policy
+			deployment := tmpl.EvaluateOrFail(t, file.AsStringOrFail(t, "testdata/config.yaml"),
```
Comment (incfly) [inline tests/integration/security/sds_vault_flow/sds_vault_flow_test.go]: optional, global-mtls.yaml seesm more illustrative than config.yaml right now. `vault-service-entry.yaml` might be a better name.

File: tests/integration/security/sds_vault_flow/sds_vault_flow_test.go [hunk 3]
```diff
@@ -85,16 +85,17 @@ func TestSdsVaultCaFlow(t *testing.T) {
...
-			g.ApplyConfigOrFail(t, systemNS, deployment)
-			defer g.DeleteConfigOrFail(t, systemNS, deployment)
 
-			// Sleep 3 seconds for the policy to take effect.
-			time.Sleep(3 * time.Second)
+			g.ApplyConfigOrFail(t, ns, deployment)
+			defer g.DeleteConfigOrFail(t, ns, deployment)
+
+			// Sleep 10 seconds for the policy to take effect.
+			time.Sleep(10 * time.Second)
```
Comment (incfly) [inline tests/integration/security/sds_vault_flow/sds_vault_flow_test.go]: same here why 10 seconds?

File: tests/integration/security/sds_citadel_flow/sds_citadel_flow_test.go (hunk 1 above)
Comment (lei-tang) [reply tests/integration/security/sds_citadel_flow/sds_citadel_flow_test.go]: 3 seconds may not be enough for a policy to take effect. 10 seconds gives more time for a policy to take effect.

File: tests/integration/security/sds_vault_flow/sds_vault_flow_test.go (hunk 3 above)
Comment (lei-tang) [reply tests/integration/security/sds_vault_flow/sds_vault_flow_test.go]: 3 seconds may not be enough for a policy to take effect. 10 seconds gives more time for a policy to take effect.

File: tests/integration/security/sds_vault_flow/sds_vault_flow_test.go (hunk 2 above)
Comment (lei-tang) [reply tests/integration/security/sds_vault_flow/sds_vault_flow_test.go]: The file name is intentionally made generic for extensibility: new policies may be placed in the config.yaml.

**Comment by lei-tang** [comment]
https://github.com/istio/istio.io/pull/4200

**Pull request merged**